tests/unit_tests/test_oedisi_tools/test_multi_container.py

```diff
@@ -0,0 +1,91 @@
+from oedisi.types.common import BROKER_SERVICE, API_FILE, HeathCheck
+from fastapi.testclient import TestClient
+from click.testing import CliRunner
+from oedisi.tools import cli
+from pathlib import Path
+import importlib
+import pytest
+import sys
+import os
+
+IN_GITHUB_ACTIONS = os.getenv("GITHUB_ACTIONS") == "true"
+
+@pytest.fixture
+def base_path() -> Path:
+    """Get the current folder of the test"""
+    return Path(__file__).parent
+
+@pytest.fixture
+def test_mc_build(base_path: Path, monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.chdir(base_path)
+    runner = CliRunner()
+    
+    broker_path = base_path / BROKER_SERVICE
+    assert broker_path.exists(), "Broker federate should be implemented before building a multicontainer problem."
+
+    api_implementation = broker_path / API_FILE
+    assert api_implementation.exists(), f"A valid REST API implementatiion should exist in {api_implementation} before building a multicontainer problem."    
+  
+    requirements_file = broker_path / "requirements.txt"
+    assert requirements_file.exists(), f"All components should have a valid requirements.txt file listing required python packages for the build."    
+              
+    result = runner.invoke(cli, ['build', "-m"])
+    assert result.exit_code == 0
+    
+@pytest.mark.usefixtures('test_mc_build')
+def test_api_heath_endpoint(base_path: Path, monkeypatch: pytest.MonkeyPatch):
+    build_path = base_path / "build"
+    assert build_path.exists(), "Build path for the test project does not exist."
+    for folder in build_path.iterdir():
+        if folder.is_dir() and folder.name not in ["kubernetes", 'tester']:
+            assert (folder/"server.py").exists(), f"Server.py does not exist for path {folder}"
+            sys.path.insert(1, str(folder.absolute()))
```

Comment:
Does this not work with monkeypatch instead of sys.path edits?

Reply:
@josephmckinsey  I have never used monkeypatch before. would you be able to refactor this?